newIDE/app/src/Course/LockedCourseChapterPreview.js

```diff
@@ -0,0 +1,218 @@
+// @flow
+
+import * as React from 'react';
+import { Trans } from '@lingui/macro';
+
+import type {
+  CourseChapter,
+  LockedVideoBasedCourseChapter,
+  LockedTextBasedCourseChapter,
+} from '../Utils/GDevelopServices/Asset';
+import Text from '../UI/Text';
+import { ColumnStackLayout, ResponsiveLineStackLayout } from '../UI/Layout';
+import Paper from '../UI/Paper';
+import RaisedButton from '../UI/RaisedButton';
+import FlatButton from '../UI/FlatButton';
+import { useResponsiveWindowSize } from '../UI/Responsive/ResponsiveWindowMeasurer';
+import { SubscriptionSuggestionContext } from '../Profile/Subscription/SubscriptionSuggestionContext';
+import GoldCompact from '../Profile/Subscription/Icons/GoldCompact';
+import Coin from '../Credits/Icons/Coin';
+import Lock from '../UI/CustomSvgIcons/Lock';
+import Window from '../Utils/Window';
+import PasswordPromptDialog from '../AssetStore/PasswordPromptDialog';
+
+const getYoutubeVideoIdFromUrl = (youtubeUrl: ?string): ?string => {
+  if (!youtubeUrl || !youtubeUrl.startsWith('https://youtu.be/')) return null;
+
+  try {
+    const url = new URL(youtubeUrl);
+
+    const lastPartOfUrl = url.pathname.split('/').pop();
+    if (!lastPartOfUrl || !lastPartOfUrl.length) {
+      console.error(`The video URL is badly formatted ${youtubeUrl}`);
+      return null;
+    }
+    return lastPartOfUrl;
+  } catch (error) {
+    console.error(`Could not parse youtube url ${youtubeUrl}:`, error);
+    return null;
+  }
+};
+
+const styles = {
+  videoAndMaterialsContainer: {
+    display: 'flex',
+    marginTop: 8,
+    gap: 8,
+    alignItems: 'stretch',
+    flexWrap: 'wrap',
+    marginBottom: 8,
+    flex: 1,
+    minWidth: 0,
+  },
+  lockedChapterTextContainer: {
+    // Similar to Line component but without the minWidth:0 that somehow
+    // prevents container to wrap when overflowing.
+    display: 'flex',
+    flex: 1,
+    minHeight: 0,
```

Comment:
I think the prop `useFullHeight` of the Line component does the same thing (plus your comm mentions minWidth instead of minheight)